From f6e0127a9da1c2dd3118304c74d59de7a859f12a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:16 -0400
Subject: [tools] perf: Generalize percent_color_snprintf()

Message-id: <1408953210-26343-249-git-send-email-jolsa@redhat.com>
Patchwork-id: 88881
O-Subject: [PATCH RHEL7.1 BZ1133083 248/322] perf tools: Generalize percent_color_snprintf()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit f77c6e9c8f9c444cd44423df0c2708e86a06a696
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Mon Dec 30 13:04:18 2013 +0530

description
===========
Make percent_color_snprintf() handle negative values correctly.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/color.c b/tools/perf/util/color.c
index 66e44a5..8cda46c 100644
--- a/tools/perf/util/color.c
+++ b/tools/perf/util/color.c
@@ -1,6 +1,7 @@
 #include <linux/kernel.h>
 #include "cache.h"
 #include "color.h"
+#include <math.h>
 
 int perf_use_color_default = -1;
 
@@ -298,10 +299,10 @@ const char *get_percent_color(double percent)
   * entries in green - and keep the low overhead places
   * normal:
   */
- if (percent >= MIN_RED)
+ if (fabs(percent) >= MIN_RED)
   color = PERF_COLOR_RED;
  else {
-  if (percent > MIN_GREEN)
+  if (fabs(percent) > MIN_GREEN)
    color = PERF_COLOR_GREEN;
  }
  return color;
-- 
1.7.1